File: RedBlackTrees/java/Node.java
package RedBlackTree;

public class Node<T extends Comparable<T>> {
	private T data;
	private Color color;
	private Node<T> leftNode;
	private Node<T> rightNode;
	private Node<T> parent;

	public Node (T data){
		this.data = data;
		this.color = Color.RED;
	}

	public T getData() {
		return data;
	}

	public void setData(T data) {
		this.data = data;
	}

	public Color getColor() {
		return color;
	}

	public void setColor(Color color) {
		this.color = color;
	}

	public Node<T> getLeftNode() {
		return leftNode;
	}

	public void setLeftNode(Node<T> leftNode) {
		this.leftNode = leftNode;
	}

	public Node<T> getRightNode() {
		return rightNode;
	}

	public void setRightNode(Node<T> rightNode) {
		this.rightNode = rightNode;
	}

	public Node<T> getParent() {
		return parent;
	}

	public void setParent(Node<T> parent) {
		this.parent = parent;
	}
}

File: BinarySearchTrees/cpp/Node.cpp
#include "Node.h"

template <typename T>
Node<T>::Node(T data){
	this->leftNode = nullptr;
	this->data = data;
	this->rightNode = nullptr;
}

template <typename T>
T Node<T>::getData(){
	return this->data;
}

template <typename T>
void Node<T>::setData(T data){
	this->data = data;
}

template <typename T>
Node<T>* Node<T>::getLeftNode(){
	return this->leftNode;
}

template <typename T>
void Node<T>::setLeftNode(Node<T> *node){
	this->leftNode = node;
}

template <typename T>
Node<T>* Node<T>::getRightNode(){
	return this->rightNode;
}

template <typename T>
void Node<T>::setRightNode(Node<T> *node){
	this->rightNode = node;
}